src/libraries/System.Net.Requests/src/System/Net/HttpWebRequest.cs

```diff
@@ -1666,6 +1672,13 @@ private static HttpClient CreateHttpClient(HttpClientParameters parameters, Http
                 handler.Expect100ContinueTimeout = parameters.ContinueTimeout;
                 client.Timeout = parameters.Timeout;
 
+                if (request != null && request.ImpersonationLevel != TokenImpersonationLevel.None)
+                {
+                    // This is legacy hack. We want to deal with it only if explicitly set.
+                    var setting = typeof(SocketsHttpHandler).InvokeMember("_settings", BindingFlags.GetField | BindingFlags.NonPublic | BindingFlags.Instance, null, handler, null);
+                    setting?.GetType().InvokeMember("_impersonationLevel", BindingFlags.SetField | BindingFlags.NonPublic | BindingFlags.Public | BindingFlags.Instance, null, setting, new object[] { request!.ImpersonationLevel });
```

Comment:
Should we be add something public to SocketsHttpHandler instead of using private reflection?

Reply:
This is what https://github.com/dotnet/runtime/issues/100895 is asking for. My hesitation comes from
1) it works only on Windows
2) It is somewhat deprecated and it no longer works by default in new browsers e.g. IE was fine and Edge is not. 

So my primary concern was support for legacy app migration. 
If we see value, adding public property would not be too difficult.  
Let me know @stephentoub and we can push https://github.com/dotnet/runtime/issues/100895 forward for `SocketsHttpHandler`. 

We still _may_ proceed with this so testing is possible and make updates once public API is available.

Reply:
> We still may proceed with this so testing is possible and make updates once public API is available.

Ok. I don't love the private reflection, but as the components are both in the shared framework and versioned together, I can tolerate it.

> it works only on Windows, It is somewhat deprecated and it no longer works by default in new browsers e.g. IE was fine and Edge is not.

Ok, then I'd agree it's not desirable to expose.

Reply:
https://learn.microsoft.com/en-us/troubleshoot/developer/webapps/iis/www-authentication-authorization/kerberos-double-hop-authentication-edge-chromium

we can revisit public API if more desirable for whatever reason.